CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -115,12 +206,84 @@ public async Task GetStudentsResult()
                 }
             };
 
+            var expectedStudentsMarks = new List<AverageStudentMarkDto>
+            {
+                new AverageStudentMarkDto()
+                    {
+                        CourseId = 2,
+                        StudentGroupId = 2,
+                        StudentId = 6,
+                        StudentAverageMark = (decimal)5.1
+                    }
+            };
+
+            var expectedAverageStudentVisits = new List<AverageStudentVisitsDto>
+            {
+                new AverageStudentVisitsDto()
+                    {
+                        CourseId = 2,
+                        StudentGroupId = 2,
+                        StudentId = 6,
+                        StudentAverageVisitsPercentage = 15
+                    }
+            };
+
+
+            var expectedStudentsResult = new StudentsResultsDto()
+            {
+                AverageStudentVisits = expectedAverageStudentVisits,
+                AverageStudentsMarks = expectedStudentsMarks
+            };
+
+            _dashboardRepositoryMock.Setup(x => x.GetGroupsIdsByCourseIdAndPeriodAsync(
+                (long)studentResultRequestWithData.CourseId,
+                studentResultRequestWithData.StartDate,
+                studentResultRequestWithData.FinishDate))
+                .ReturnsAsync(new List<long>() { 2 });
+
+            _dashboardRepositoryMock.Setup(x => x.GetStudentsIdsByGroupIdsAsync(new List<long>() { 2 }))
+                .ReturnsAsync(new List<long>() { 6, 7, 8, 9, 10 });
+
+            _dashboardRepositoryMock.Setup(x => x.GetStudentAverageMarksByStudentIdsAndGropsIdsAsync(
+              new List<long>() { 6, 7, 8, 9, 10 }, new List<long>() { 2 }))
+               .ReturnsAsync(expectedStudentsMarks);
+
+            _dashboardRepositoryMock.Setup(x => x.GetStudentsAverageVisitsByStudentIdsAndGroupsIdsAsync(
+                 new List<long>() { 6, 7, 8, 9, 10 }, new List<long>() { 2 }))
+                .ReturnsAsync(expectedAverageStudentVisits);
```

Comment:
Perhaps we could declare lists of student group ids beforehand to avoid using recurring magical numbers
```cs
var existentGroupIds = new List<long>() { 6, 7, 8, 9, 10 };
var expectedStudentAverageMarks = new List<long>() { 2 };
// etc
```